api/controllers/UserController.js

```diff
@@ -234,6 +235,301 @@ module.exports = {
     });
   },
 
+  permaBan: function (req, res) {
+    /*  Form parameters:
+          req.params.username: The user who is being banned
+          req.params.banNote: The ban reason to go on the mod log (not visible to banned user, 300 characters max)
+          req.params.banMessage: The note that gets sent with the "you are banned" PM
+          req.params.banlistEntry: The ban reason to appear on the public banlist
+          req.params.duration: The number of days that the user will be banned for.
+        Permaban process:
+          1. Ban user from /r/pokemontrades
+          2. Ban user from /r/SVExchange
+          3. Add "BANNED USER" to user's flair on /r/pokemontrades
+          4. Add user's friend code to /r/pokemontrades AutoModerator config (2 separate lists)
+          5. Add user's friend code to /r/SVExchange AutoModerator config (2 separate lists)
+          6. Remove all of the user's TSV threads on /r/SVExchange
+          7. Add user's info to banlist wiki on /r/pokemontrades
+    */
```

Comment:
To do: Make the permaban function also add the user to the local FlairHQ banlist. This prevents the user from deleting or modifying trades (e.g. to delete incriminating evidence in the event of an appeal)